### PR TITLE
Sync Java 17 changes into master

### PR DESCRIPTION
**Pull request by gabilang**
## Purpose
Java 17 migration
Fixes:

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility
- [ ] No `commons` package changes (if there are any, please update the GraphQL version in [GraphQL tools](https://github.com/ballerina-platform/graphql-tools))

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -108,7 +108,7 @@ task build {
     dependsOn(':graphql-ballerina:build')
     dependsOn(':graphql-compiler-plugin-tests:test')
     dependsOn(':graphql-ballerina-tests:test')
-    dependsOn(':graphql-examples:build')
+//    dependsOn(':graphql-examples:build')
```
Comment (ThisaruGuruge) [inline build.gradle:111]: Why the example build is omitted?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.2-bin.zip
```
Comment (ThisaruGuruge) [inline gradle/wrapper/gradle-wrapper.properties:3]: Shouldn't we re-commit the wrapper jar, if we are updating the Gradle version?

File: settings.gradle [hunk 3]
```diff
@@ -49,7 +49,7 @@ project(':graphql-compiler-plugin').projectDir = file("compiler-plugin")
 project(':graphql-ballerina').projectDir = file("ballerina")
 project(':graphql-compiler-plugin-tests').projectDir = file("compiler-plugin-tests")
 project(':graphql-ballerina-tests').projectDir = file("ballerina-tests")
-project(':graphql-examples').projectDir = file("examples")
+ project(':graphql-examples').projectDir = file("examples")
```
Comment (ThisaruGuruge) [inline settings.gradle:52]: Extra space?

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (warunalakshitha) [reply gradle/wrapper/gradle-wrapper.properties]: Updated gradle wrapper files.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**